### DISCUSSION TITLE
Definitions - speaking the same language

### DISCUSSION THREAD

**Discussion by longr** [NERC-CEH/rse_group#34] [General]
I don't like to prescribe things top down, but I wonder if as a group (EDS) it would be worth adopting some standard definitions,  at least as starting point so we are all talking the same language. 

This came from the FDRI stand-up where they had been discussing numbered data levels (I hate numbers over descriptions, but that is a personal opinion.)  We also have a similar issue when it comes to levels of reproducibility in software and what is expected at each level.  Here are three links, one to [software definitions](https://www.software.ac.uk/news/new-guide-how-reproducible-should-research-software-be), one to [data definitions](https://thetius.com/the-nuances-behind-data-definitions/), and one to [NASAs TRL](https://www.nasa.gov/directorates/somd/space-communications-navigation-program/technology-readiness-levels/):

**Comment by mattjbr123** [comment]
I really like the [software definitions](https://www.software.ac.uk/news/new-guide-how-reproducible-should-research-software-be) article, I can see myself referring back to that

**Reply by longr** [reply]
I think it is perhaps needs some more granularity, but it is definitely useful.

**Comment by mattjbr123** [comment]
I would like to know what a RESTful API is, all the articles I've read so far are a little impenetrable/don't really mean anything to me...

**Reply by matthewcoole** [reply]
Not an expert, and I'm not sure if I can give a standard definition that will be accepted by everyone (but I can give it a go). A REST API is just an API where any state will be stored in the requests themselves; state shouldn't be stored server side. This could mean using things like an access key rather than calling a login endpoint and getting a session id etc. Generally it also means responses will be consistent and not stochastic so that they can be cached to increase throughput. Beyond that I'd say they're like any other type of API, data can be returned in whatever format is most appropriate. 

I'm sure there might be some other points to it (maybe CRUD?) so anyone else is free to "correct" my definition ;)

**Reply by mattjbr123** [reply]
Thanks Matt - that definitely helps. I meant to link this post to the social coding topic I started in the new repo too: https://github.com/ukceh-rse/community/discussions/9

**Reply by longr** [reply]
Following on from this, perhaps we should start a glossary of terms, but not just dry CS descriptions, but with more examples and potentially whys (you would use it).   For instance I had to look up *serialisation* again and the descriptions made no sense until I found one with an example and reason why you would use it.  REST API fits with this I suspect.  CRUD and ETL perhaps need less of an example, but might be nice to create a glossary.

**Reply by mattjbr123** [reply]
Strongly in favour! I learn best by examples for sure
